core/primitives/src/merkle.rs

```diff
@@ -22,10 +22,8 @@ pub enum Direction {
     Right,
 }
 
-pub fn combine_hash(hash1: MerkleHash, hash2: MerkleHash) -> MerkleHash {
-    let mut combined: Vec<u8> = hash1.into();
-    combined.append(&mut hash2.into());
-    hash(&combined)
+pub fn combine_hash(hash1: &MerkleHash, hash2: &MerkleHash) -> MerkleHash {
```

Comment:
did you verify that it generates the same results as before? can you add a small test here, that would compute the combine_hash - and compare to a fixed result (and make sure that this result is the same before and after your PR)

Reply:
@mm-near Yes, I did. I included a test in this PR.